src/app/Header.js

```diff
@@ -53,12 +55,44 @@ function NavLink({link}) {
   )
 }
 
+function NavLinkAuth() {
+  const [authToken, setAuthToken] = useState(null);
+  const mutation = useMutation(logout, {
+    onSuccess: () => {
+      localStorage.removeItem(AUTH_TOKEN_KEY);
+      localStorage.removeItem(USER_ID_KEY);
+      localStorage.removeItem(SIGNUP_CURRENT_STEP_KEY);
+    },
+    onError: () => {
+      console.error("Logout failed");
+    }
+  });
+
+  const theme = useTheme()
+
+  useEffect(() => {
+    setAuthToken(localStorage.getItem(AUTH_TOKEN_KEY));
+  }, []);
+
+  return (
+    <NextLink href="/" passHref>
+      <Button variant="outlined" color="orange" onClick={authToken ? mutation.mutate : null} sx={{
```

Comment:
Why do we read initial authtoken from state?
User logged in? Read auth token from localStorage, delete it.
User not logged in? Read auth token from state, do nothing.

Reply:
1. it is for using useEffect for getting it (solve problem with server rendering)

Reply:
Do you want to show Log out botton in any case?